adlfs/tests/test_spec.py

```diff
@@ -2045,3 +2046,82 @@ def test_open_file_x(storage: azure.storage.blob.BlobServiceClient, tmpdir):
         with fs.open("data/afile", "xb") as f:
             pass
     assert fs.cat_file("data/afile") == b"data"
+
+
+def test_number_of_blocks(storage, mocker):
+    from azure.storage.blob.aio import BlobClient
+
+    blocksize = 5 * 2**20
+    fs = AzureBlobFileSystem(
+        account_name=storage.account_name,
+        connection_string=CONN_STR,
+        blocksize=blocksize,
+    )
+
+    content = b"1" * (blocksize * 4 + 1)
+    with fs.open("data/root/a/file.txt", "wb") as f:
+        mock_stage_block = mocker.patch.object(BlobClient, "stage_block")
+        mock_commit_block_list = mocker.patch.object(BlobClient, "commit_block_list")
+        f.write(content)
+        expected_blocks = math.ceil(len(content) / blocksize)
+        actual_blocks = mock_stage_block.call_count
+        assert actual_blocks == expected_blocks
+        block_lengths = [
+            call.kwargs["length"] for call in mock_stage_block.call_args_list
+        ]
+        assert sum(block_lengths) == len(content)
+
+    assert len(mock_commit_block_list.call_args.kwargs["block_list"]) == expected_blocks
+
+
+@pytest.mark.parametrize(
+    "filesystem_blocksize, file_blocksize, expected_blocksize",
```

Comment:
It's also probably worth adding an `expected_filesystem_blocksize` to make the blocksize are expected at the filesystem level as well in this test case.